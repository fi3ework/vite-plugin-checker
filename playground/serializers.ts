import os from 'os'

const winNewLineReg = /\/r\/n/gim
const winSepReg = /\\/g

function doesUseDoubleSlashAsPath(val: string) {
  return val.includes('//vite-plugin-checker//')
}

export const normalizeWindowsLogSerializer = {
  print(val: string, serialize) {
    let result = val
    if (os.platform() === 'win32') {
      result = result.replaceAll(winNewLineReg, '/n')
      result = result.replaceAll(process.cwd().replace(winSepReg, '/'), '<PROJECT_ROOT>')

      if (doesUseDoubleSlashAsPath(result)) {
        result = result.replaceAll(
          `//a//vite-plugin-checker//vite-plugin-checker//playground-temp`,
          '<PROJECT_ROOT>/playground-temp'
        )
        result = result.split('//').join('/')
      }
    }

    return serialize(result)
  },
  test(val) {
    if (typeof val !== 'string') return false

    if (
      (os.platform() === 'win32' &&
        (val.includes(process.cwd().replace(winSepReg, '/')) || winNewLineReg.test(val))) ||
      doesUseDoubleSlashAsPath(val)
    ) {
      return true
    }

    return false
  },
}
