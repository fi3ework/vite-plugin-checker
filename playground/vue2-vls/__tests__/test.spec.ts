import stringify from 'fast-json-stable-stringify'

import {
  killServer,
  preTest,
  resetReceivedLog,
  sleepForEdit,
  sleepForServerReady,
  stripedLog,
  viteBuild,
  viteServe,
} from '../../../packages/vite-plugin-checker/__tests__/e2e/Sandbox/Sandbox'
import {
  editFile,
  sleep,
  testDir,
  WORKER_CLEAN_TIMEOUT,
} from '../../../packages/vite-plugin-checker/__tests__/e2e/testUtils'
import { copyCode } from '../../../scripts/jestSetupFilesAfterEnv'
import { serializers } from '../../../scripts/serializers'

expect.addSnapshotSerializer(serializers)

beforeAll(async () => {
  await preTest()
})

afterAll(async () => {
  await sleep(WORKER_CLEAN_TIMEOUT)
})

describe('vue2-vls', () => {
  beforeEach(async () => {
    await copyCode()
  })

  describe('serve', () => {
    afterEach(async () => {
      await killServer()
    })

    it('get initial error and subsequent error', async () => {
      let err: any

      await viteServe({
        cwd: testDir,
        // @ts-expect-errorÆ’
        wsSend: (_payload) => (err = _payload.err),
      })

      await sleepForServerReady(2)
      expect(stringify(err)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()

      console.log('-- edit file --')
      resetReceivedLog()
      editFile('src/components/HelloWorld.vue', (code) => code.replace('msg1', 'msg2'))
      await sleepForEdit()

      expect(stringify(err)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()
    })
  })

  describe('build', () => {
    beforeEach(async () => {
      await copyCode()
    })

    it('enableBuild: true', async () => {
      await viteBuild({
        expectedErrorMsg: `Property 'msg1' does not exist on type`,
        cwd: testDir,
      })
    })

    it('enableBuild: false', async () => {
      editFile('vite.config.ts', (code) =>
        code.replace('checker({ vls: {} })', 'checker({ vls: {}, enableBuild: false })')
      )

      await viteBuild({
        unexpectedErrorMsg: `Property 'msg1' does not exist on type`,
        cwd: testDir,
      })
    })
  })
})
