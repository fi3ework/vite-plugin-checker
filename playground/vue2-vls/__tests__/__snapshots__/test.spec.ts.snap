// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue2-vls serve get initial error and subsequent error 1`] = `"[{\\"frame\\":\\"  1 | <template>/n  2 |   <div class=/\\"hello/\\">/n> 3 |     <h1>{{ msg1 }}</h1>/n    |            ^^^^/n  4 |     <p>/n  5 |       For a guide and recipes on how to configure / customize this project,<br />/n  6 |       check out the\\",\\"id\\":\\"<PROJECT_ROOT>/temp/vue2-vls/src/components/HelloWorld.vue\\",\\"loc\\":{\\"column\\":12,\\"file\\":\\"<PROJECT_ROOT>/temp/vue2-vls/src/components/HelloWorld.vue\\",\\"line\\":3},\\"message\\":\\"Property 'msg1' does not exist on type 'CombinedVueInstance<{ msg: string; } & Vue, object, object, object, Record<never, any>>'. Did you mean 'msg'?\\",\\"plugin\\":\\"VLS\\",\\"stack\\":\\"\\"}]"`;

exports[`vue2-vls serve get initial error and subsequent error 2`] = `
" ERROR(VLS)  Property 'msg1' does not exist on type 'CombinedVueInstance<{ msg: string; } & Vue, object, object, object, Record<never, any>>'. Did you mean 'msg'?
 FILE  <PROJECT_ROOT>/temp/vue2-vls/src/components/HelloWorld.vue:3:12

  1 | <template>
  2 |   <div class=\\"hello\\">
> 3 |     <h1>{{ msg1 }}</h1>
    |            ^^^^
  4 |     <p>
  5 |       For a guide and recipes on how to configure / customize this project,<br />
  6 |       check out the
"
`;

exports[`vue2-vls serve get initial error and subsequent error 3`] = `"[{\\"frame\\":\\"  1 | <template>/n  2 |   <div class=/\\"hello/\\">/n> 3 |     <h1>{{ msg2 }}</h1>/n    |            ^^^^/n  4 |     <p>/n  5 |       For a guide and recipes on how to configure / customize this project,<br />/n  6 |       check out the\\",\\"id\\":\\"<PROJECT_ROOT>/temp/vue2-vls/src/components/HelloWorld.vue\\",\\"loc\\":{\\"column\\":12,\\"file\\":\\"<PROJECT_ROOT>/temp/vue2-vls/src/components/HelloWorld.vue\\",\\"line\\":3},\\"message\\":\\"Property 'msg2' does not exist on type 'CombinedVueInstance<{ msg: string; } & Vue, object, object, object, Record<never, any>>'. Did you mean 'msg'?\\",\\"plugin\\":\\"VLS\\",\\"stack\\":\\"\\"}]"`;

exports[`vue2-vls serve get initial error and subsequent error 4`] = `
" ERROR(VLS)  Property 'msg2' does not exist on type 'CombinedVueInstance<{ msg: string; } & Vue, object, object, object, Record<never, any>>'. Did you mean 'msg'?
 FILE  <PROJECT_ROOT>/temp/vue2-vls/src/components/HelloWorld.vue:3:12

  1 | <template>
  2 |   <div class=\\"hello\\">
> 3 |     <h1>{{ msg2 }}</h1>
    |            ^^^^
  4 |     <p>
  5 |       For a guide and recipes on how to configure / customize this project,<br />
  6 |       check out the
"
`;
